samples/wasm-c-api/CMakeLists.txt

```diff
@@ -87,6 +87,16 @@ endif()
 set(WAMR_ROOT_DIR ${CMAKE_CURRENT_LIST_DIR}/../..)
 include (${WAMR_ROOT_DIR}/build-scripts/runtime_lib.cmake)
 
+if (NOT DEFINED SANITIZER)
+  set(SANITIZER "")
+endif()
+
+if (SANITIZER STREQUAL "ubsan")
+  set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -g -O2 -fno-omit-frame-pointer -fsanitize=undefined -fno-sanitize-recover=all -fno-sanitize=alignment" )
```

Comment:
Do we also need to check for a compiler here, or are we ok with the compiler (sliently) failing if those flags are not supported?

Reply:
I believe that you won't set these flags for the wrong the wrong compiler on the CI side. 

But if you think it's better add here I'll add